cmd/cloud/dbt.go

```diff
@@ -119,19 +111,91 @@ func deployDbt(cmd *cobra.Command, args []string) error {
 
 	// deploy the dbt project as a bundle
 	deployBundleInput := &cloud.DeployBundleInput{
-		BundlePath:   dbtProjectPath,
-		MountPath:    mountPath,
-		DeploymentID: deploymentID,
-		BundleType:   dbtBundleType,
-		Description:  deployDescription,
-		Wait:         waitForDeploy,
-	}
-	err = DeployBundle(deployBundleInput, platformCoreClient, astroCoreClient)
+		BundlePath:         dbtProjectPath,
+		MountPath:          mountPath,
+		DeploymentID:       deploymentID,
+		BundleType:         dbtBundleType,
+		Description:        deployDescription,
+		Wait:               waitForDeploy,
+		PlatformCoreClient: platformCoreClient,
+		CoreClient:         astroCoreClient,
+	}
+	return DeployBundle(deployBundleInput)
+}
+
+func newDbtDeleteCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete DEPLOYMENT-ID",
+		Short: "Delete a dbt project from a Deployment on Astro",
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  deleteDbt,
+		Example: `
+Specify the ID of the Deployment on Astronomer you would like to delete this dbt project from:
+
+  $ astro dbt delete <deployment ID>
+
+Menu will be presented if you do not specify a deployment ID:
+
+  $ astro dbt delete
+`,
+	}
+
+	cmd.Flags().StringVarP(&mountPath, "mount-path", "m", "", fmt.Sprintf("Path to mount dbt project in Airflow, for reference by DAGs. Default %s{dbt project name}", dbtDefaultMountPathPrefix))
+	cmd.Flags().StringVarP(&dbtProjectPath, "project-path", "p", "", "Path to the dbt project to delete from the Deployment. Default current directory")
+	cmd.Flags().StringVar(&workspaceID, "workspace-id", "", "Workspace for your Deployment")
+	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "Name of the Deployment to deploy to")
+	cmd.Flags().StringVarP(&deployDescription, "description", "", "", "Description to store on the deploy")
+	cmd.Flags().BoolVarP(&waitForDeploy, "wait", "w", false, "Wait for the Deployment to become healthy before ending the command")
+
+	return cmd
+}
+
+func deleteDbt(cmd *cobra.Command, args []string) error {
+	cmd.SilenceUsage = true
+
+	// if the workspace ID is not provided, try to find a valid workspace
+	if workspaceID == "" {
+		var err error
+		workspaceID, err = coalesceWorkspace()
+		if err != nil {
+			return fmt.Errorf("failed to find a valid workspace: %w", err)
+		}
+	}
+
+	// get the deployment id to deploy the dbt project to
+	deploymentID, err := resolveDeploymentIDFromDbtArgsFlags(args, workspaceID, deploymentName)
 	if err != nil {
 		return err
 	}
+	fmt.Println("Initiating dbt delete deploy for deployment ID: " + deploymentID)
 
-	return nil
+	// if the mount path is not provided, derive it from the dbt project name
+	if mountPath == "" {
+		// extract the dbt project's name
+		dbtProjectName, err := extractDbtProjectName(dbtProjectPath)
```

Comment:
general questions: 
Do we feel not having any form of identifier for the deploy bundle makes it harder to maintain dbt projects for the users? In this particular case, the user needs to have the project at `dbtProjectPath` to delete the remote bundle (which feels a bit weird).

Also, otherwise, would users be able to figure out mountPath for each deploy bundle in an Airflow deployment, so that they can provide it in this command?

Reply:
If we generated an identifier I think that would have the same discoverability problem as the mount path. It might be a good idea for us to also have a CLI command that lists the dbt projects on the deployment, so that the user can see what is present and what mount path would be used to remove one (in case the dbt project itself is not locally accessible).

Reply:
I think in general having a low level thing like a mount path be the key would be a bit odd, but that is the actual value that users will have to plug into their DAGs to use the files. e.g. in the Cosmos `ProjectConfig` https://astronomer.github.io/astronomer-cosmos/getting_started/astro.html#create-a-dagfile

Reply:
> It might be a good idea for us to also have a CLI command that lists the dbt projects on the deployment, so that the user can see what is present and what mount path would be used to remove one

True, that makes sense

> but that is the actual value that users will have to plug into their DAGs to use the files. e.g. in the Cosmos ProjectConfig https://astronomer.github.io/astronomer-cosmos/getting_started/astro.html#create-a-dagfile

ahh, got it, that's solves my concern to some extent at least. 🙂